src/main/java/uk/gov/pay/ledger/expungeorredact/service/ExpungeOrRedactService.java

```diff
@@ -118,27 +119,24 @@ private void redactPIIFromTransactionsAndDeleteRelatedEvents() {
 
         noOfTransactionsRedactedMetric.observe(noOfTxsProcessed);
         noOfEventsDeletedRedactedMetric.observe(noOfEventsDeleted);
-
-        if (noOfTxsProcessed > 0) {
-            transactionRedactionInfoDao.update(createdDateOfLastProcessedTransaction);
-        }
     }
 
     private ZonedDateTime getRedactTransactionsUpToDate() {
         return clock.instant()
-                .minus(expungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays(), ChronoUnit.DAYS)
+                .minus(expungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays(), DAYS)
                 .atZone(UTC);
     }
 
     private ZonedDateTime getCreatedDateOfLastProcessedTransaction() {
         return ofNullable(transactionRedactionInfoDao.getCreatedDateOfLastProcessedTransaction())
-                .orElseGet(this::getCreatedDateOfFirstTransaction)
+                .orElseGet(this::getDateBeforeCreatedDateOfFirstTransaction)
                 .withZoneSameInstant(UTC);
     }
 
-    private ZonedDateTime getCreatedDateOfFirstTransaction() {
+    private ZonedDateTime getDateBeforeCreatedDateOfFirstTransaction() {
```

Comment:
Should this be `getDayBeforeCreatedDateOfFirstTransaction` instead? Also why is this being changed to the day before the created date of the first transaction?

Reply:
Updated name! 

DAO returns transaction **after** the last transaction processed date https://github.com/alphagov/pay-ledger/pull/2923/files#diff-257f1e610732a7f40031675387f1da0980f7c707f2704b901e8d87b1411bb6caR89 so for the first run this should be before the day so first transaction is included in the result.